buildkite/resource_cluster_queue_test.go

```diff
@@ -0,0 +1,218 @@
+					testAccCheckClusterQueueRemoteValues(&cq, "Acceptance Test bar", "foobar"),
+					// Confirm the cluster queue has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_cluster_queue.foobar", "description", "Acceptance Test bar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccClusterQueue_import(t *testing.T) {
+	var cq ClusterQueueResourceModel
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testAccCheckClusterQueueDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccClusterQueueConfigBasic("foo"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the cluster queue exists in the buildkite API
+					testAccCheckClusterQueueExists("buildkite_cluster_queue.foobar", &cq),
+					// Check to confirm the local state is correct before we re-import it
+					resource.TestCheckResourceAttr("buildkite_cluster_queue.foobar", "key", "foobar"),
+					resource.TestCheckResourceAttr("buildkite_cluster_queue.foobar", "description", "Acceptance Test foo"),
+				),
+			},
+			{
+				// re-import the resource (using the graphql token of the existing resource) and confirm they match
+				ResourceName:      "buildkite_cluster_queue.foobar",
+				ImportStateIdFunc: testAccGetImportClusterQueueId(&cq),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckClusterQueueExists(resourceName string, clusterQueueResourceModel *ClusterQueueResourceModel) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resourceState, ok := s.RootModule().Resources[resourceName]
+
+		if !ok {
+			return fmt.Errorf("Not found in state: %s", resourceName)
+		}
+
+		if resourceState.Primary.ID == "" {
+			return fmt.Errorf("No ID is set in state")
+		}
+
+		// Obtain queues of the queue's cluster from its cluster UUID
+		queues, err := getClusterQueues(
+			genqlientGraphql,
+			getenv("BUILDKITE_ORGANIZATION_SLUG"),
+			resourceState.Primary.Attributes["cluster_uuid"],
+		)
+
+		// If cluster queues were not able to be fetched by Genqlient
+		if err != nil {
+			return fmt.Errorf("Error fetching Cluster queues from graphql API: %v", err)
+		}
+
+		// Obtain the ClusterQueueResourceModel from the queues slice
+		for _, edge := range queues.Organization.Cluster.Queues.Edges {
+			if edge.Node.Id == resourceState.Primary.ID {
+				updateClusterQueueResource(edge.Node, clusterQueueResourceModel)
+				break
+			}
+		}
+
+		// If clusterQueueResourceModel isnt set from the queues slice
+		if clusterQueueResourceModel.Id.ValueString() == "" {
+			return fmt.Errorf("No Cluster queue found with graphql id: %s", resourceState.Primary.ID)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckClusterQueueRemoteValues(cq *ClusterQueueResourceModel, description string, key string) resource.TestCheckFunc {
```

Comment:
When 2 parameters share a type you can declare the type once, ie:

```go
func testAccCheckClusterQueueRemoteValues(cq *ClusterQueueResourceModel, description, key string) resource.TestCheckFunc {
```

Reply:
👍